tests/tensor_indexing.rs

```diff
@@ -0,0 +1,114 @@
+use tch::{Device, Kind, Tensor};
+use tch::{IndexOp, NewAxis};
+
+#[test]
+fn integer_index() {
+    let opt = (Kind::Float, Device::Cpu);
+
+    let tensor = Tensor::arange1(0, 2 * 3, opt).view(&[2, 3]);
+    let result = tensor.i(1);
+    let expect = Tensor::arange1(3, 6, opt);
+    assert!(result.eq1(&expect).all().int64_value(&[]) == 1);
```

Comment:
When comparing the result tensor to a 1-d tensor, I find the following to be slightly more readable:
`assert_eq!(Vec::<i64>::from(&result), [3, 4, 5]);`
(the same applies to further tests below)

Reply:
I would also add an `assert_eq!(result.shape, [3])` just in case.
(the same applies to further tests below, maybe when the result is not a 1d tensor you can assert both on shape + on the flattened version?)